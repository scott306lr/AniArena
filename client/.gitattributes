# This way Git won't bother showing massive diffs when each time you subsequently add or update them:
/.yarn/releases/** binary
/.yarn/plugins/** binary