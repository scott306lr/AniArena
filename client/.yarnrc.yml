nodeLinker: node-modules

yarnPath: .yarn/releases/yarn-3.2.2.cjs

packageExtensions:
  # firebase warnings fix
  "@firebase/analytics-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"

  "@firebase/app-check-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"

  "@firebase/auth-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"
      "@firebase/app-types": "0.x"

  "@firebase/database@0.x":
    peerDependencies:
      "@firebase/app-types": "0.x"

  "@firebase/database-compat@0.x":
    peerDependencies:
      "@firebase/app-types": "0.x"

  "@firebase/firestore-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"
      "@firebase/app-types": "0.x"
  
  "@firebase/functions@0.x":
    peerDependencies:
      "@firebase/app-types": "0.x"

  "@firebase/functions-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"
      "@firebase/app-types": "0.x"

  "@firebase/installations-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"
      "@firebase/app-types": "0.x"

  "@firebase/messaging-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"

  "@firebase/performance-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"

  "@firebase/remote-config-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"

  "@firebase/storage-compat@0.x":
    peerDependencies:
      "@firebase/app": "0.x"
      "@firebase/app-types": "0.x"

