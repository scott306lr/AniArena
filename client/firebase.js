// Import the functions you need from the SDKs you need
import { initializeApp, getApps, getApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getStorage } from "firebase/storage";

import { getAnalytics } from "firebase/analytics";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries 

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
  storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MSG_SENDER_ID,
  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
  measurementId: process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID
};

// Initialize Firebase
// Singleton pattern: insure there's only one app while SSR
const app = !getApps().length ? initializeApp(firebaseConfig) : getApp();
const db = getFirestore();
const storage = getStorage();

export { app, db, storage };
// const analytics = getAnalytics(app);