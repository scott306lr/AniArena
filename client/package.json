{
  "private": true,
  "scripts": {
    "codegen": "graphql-codegen -r ts-node/register",
    "dev": "yarn codegen && next",
    "build": "yarn codegen && next build",
    "test": "yarn codegen && jest",
    "start": "next start"
  },
  "dependencies": {
    "@apollo/client": "^3.5.10",
    "@babel/core": "^7.18.9",
    "@babel/preset-env": "^7.18.9",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@graphql-tools/load-files": "6.0.18",
    "@graphql-tools/merge": "6.0.18",
    "@graphql-tools/schema": "6.0.18",
    "@graphql-yoga/node": "^2.2.1",
    "@heroicons/react": "^1.0.6",
    "@next-auth/firebase-adapter": "^1.0.1",
    "@types/node": "^18.0.6",
    "firebase": "^9.9.1",
    "framer-motion": "^6.5.1",
    "graphql": "15.6.0",
    "graphql-tag": "^2.12.6",
    "next": "latest",
    "next-auth": "^4.10.2",
    "react": "latest",
    "react-dom": "latest"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "^2.10.0",
    "@graphql-codegen/import-types-preset": "^2.2.2",
    "@graphql-codegen/typed-document-node": "^2.2.8",
    "@graphql-codegen/typescript": "^2.7.2",
    "@graphql-codegen/typescript-operations": "^2.3.5",
    "@graphql-codegen/typescript-resolvers": "^2.6.1",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^27.0.2",
    "@types/mocha": "^9.0.0",
    "@types/react": "^16.9.46",
    "@types/react-dom": "^16.9.8",
    "@types/react-test-renderer": "^17.0.1",
    "autoprefixer": "^10.4.7",
    "babel-jest": "27.2.5",
    "graphql-let": "^0.18.6",
    "jest": "^27.2.5",
    "postcss": "^8.4.14",
    "tailwindcss": "^3.1.6",
    "ts-node": "10.8.0",
    "typescript": "^4.7.4",
    "yaml-loader": "0.6.0"
  },
  "packageManager": "yarn@3.2.2"
}
