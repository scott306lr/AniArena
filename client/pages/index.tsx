import type { NextPage } from 'next'
import Head from 'next/head'
import RectCard from '../components/RectCard';
import WordBallon from '../components/WordBallon';

const Home: NextPage = () => {
  return (
    <div className="bg-slate-200 w-screen h-screen">
      <Head>
        <title>AniArena</title>
      </Head>

      <main className="grid">
        {/* sections */}
        <div className="flex items-center justify-center m-4 gap-8">
          {/* section 2 */}
          <div>
            <RectCard imgsrc="https://images-ext-1.discordapp.net/external/x08twzHAcDIWcqHEPgEIdfLNoIGST3tHUdti0Ww3b00/https/mudae.net/uploads/1232276/v6uL2AsAjYy0QE0-4OKy~0P98O5C.png" />
          </div>
          {/* section 3 */}
          <div className="space-y-4">
            <WordBallon text="ID: Admin" />
            <WordBallon text="Name: Admin" />
          </div>
        </div>

      </main>
    </div>
  )
}

export default Home
