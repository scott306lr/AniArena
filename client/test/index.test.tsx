import { InMemoryCache, gql } from '@apollo/client'
import React from 'react'
import Index from '../pages'

import { waitFor, render, screen } from "@testing-library/react";
import { MockedProvider } from '@apollo/client/testing'
import { useSession } from "next-auth/react";
jest.mock("next-auth/react");

const cache = new InMemoryCache()
cache.writeQuery({
  query: gql`
    query Viewer {
      viewer {
        id
        name
        status
      }
    }
  `,
  data: {
    viewer: {
      __typename: 'User',
      id: 'Baa',
      name: 'Baa',
      status: 'Healthy',
    },
  },
})

describe('Index', () => {
  it('renders the html we want', async () => {
    const mockSession = {
      expires: "1",
      user: { email: "a", name: "Delta", image: "c" },
    };

    (useSession as jest.Mock).mockReturnValue({
      data: null,//mockSession,
      status: "authenticated",
    });

    render(
      <MockedProvider cache={cache}>
        <Index />
      </MockedProvider>
    )

    // expect(component).toBeDefined()
    //expect(component.toJSON()).toBeTruthy()
    // await waitFor(() => new Promise((resolve) => setTimeout(resolve, 0)));
    expect(true).toBeTruthy()
  })
})
