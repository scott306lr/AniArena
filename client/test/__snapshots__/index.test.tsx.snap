// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders the html we want 1`] = `
<div>
  <div
    className="flex justify-center items-center max-w-6xl mx-5 my-2 lg:mx-auto"
  >
    <div
      className="flex w-min p-2 bg-white rounded-lg shadow-xl hover:scale-110 transition-all hover:translate-y-4"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="p-2 rounded-lg hover:bg-gray-500 hover:ease-in-out duration-200"
        >
          <svg
            aria-hidden="true"
            className="h-12 w-12"
            fill="none"
            stroke="currentColor"
            strokeWidth={2}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </div>
      </a>
      <a
        href="/characters"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="p-2 rounded-lg hover:bg-gray-500 hover:ease-in-out duration-200"
        >
          <svg
            aria-hidden="true"
            className="h-12 w-12"
            fill="none"
            stroke="currentColor"
            strokeWidth={2}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0zm6 3a2 2 0 11-4 0 2 2 0 014 0zM7 10a2 2 0 11-4 0 2 2 0 014 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </div>
      </a>
      <a
        href="/playerList"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="p-2 rounded-lg hover:bg-gray-500 hover:ease-in-out duration-200"
        >
          <svg
            aria-hidden="true"
            className="h-12 w-12"
            fill="none"
            stroke="currentColor"
            strokeWidth={2}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.055 11H5a2 2 0 012 2v1a2 2 0 002 2 2 2 0 012 2v2.945M8 3.935V5.5A2.5 2.5 0 0010.5 8h.5a2 2 0 012 2 2 2 0 104 0 2 2 0 012-2h1.064M15 20.488V18a2 2 0 012-2h3.064M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </div>
      </a>
      <a
        href="/battle"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="p-2 rounded-lg hover:bg-gray-500 hover:ease-in-out duration-200"
        >
          <svg
            aria-hidden="true"
            className="h-12 w-12"
            fill="none"
            stroke="currentColor"
            strokeWidth={2}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.752 11.168l-3.197-2.132A1 1 0 0010 9.87v4.263a1 1 0 001.555.832l3.197-2.132a1 1 0 000-1.664z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
            <path
              d="M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </div>
      </a>
      <a
        onClick={[Function]}
      >
        <div
          className="p-2 rounded-lg hover:bg-gray-500 hover:ease-in-out duration-200"
        >
          <svg
            aria-hidden="true"
            className="h-12 w-12"
            fill="none"
            stroke="currentColor"
            strokeWidth={2}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 16l-4-4m0 0l4-4m-4 4h14m-5 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h7a3 3 0 013 3v1"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>
  <main
    className="grid"
  >
    <div
      className="flex items-center justify-center m-8 gap-8"
    >
      <div>
        <div
          className="char-img"
        >
          <img
            className="rounded-lg"
            src="https://images-ext-1.discordapp.net/external/x08twzHAcDIWcqHEPgEIdfLNoIGST3tHUdti0Ww3b00/https/mudae.net/uploads/1232276/v6uL2AsAjYy0QE0-4OKy~0P98O5C.png"
          />
        </div>
      </div>
      <div
        className="space-y-4"
      >
        <div
          className="word-bubble"
        >
          ID: undefined
        </div>
        <div
          className="word-bubble"
        >
          NAME: undefined
        </div>
      </div>
    </div>
  </main>
</div>
`;
